src/protocols/secure_channel/RendezvousParameters.h

```diff
@@ -63,12 +63,19 @@ class RendezvousParameters
 
     bool HasPeerAddress() const { return mPeerAddress.IsInitialized(); }
     Transport::PeerAddress GetPeerAddress() const { return mPeerAddress; }
+    ByteSpan GetCSRNonce() const {return mCSRNonce;}
     RendezvousParameters & SetPeerAddress(const Transport::PeerAddress & peerAddress)
     {
         mPeerAddress = peerAddress;
         return *this;
     }
 
+    RendezvousParameters & SetCSRNonce(ByteSpan csrNonce)
```

Comment:
We can probably add `HasCSRNonce()` to this class. It'll be true if `SetCSRNonce()` was called. Otherwise false.

Reply:
ok, i added the flag  to check if CSRNonce was available on the Device instead of RendezvousParameters as we store it on the device but i guess this is also fine as i assume RendezvousParameters is designed to hold the commissioning states.  Is that accurate ?

Reply:
Should we have some comment regarding expected lifetime of this bytespan? the span is a raw pointer + size, so we need to determine how long we expect this raw pointer to point to valid data once passed to rendezvous.

Does rendezvous create a copy (then we only need it valid during init) or does it need to last longer?

Reply:
Pankaj, do you have a suggestion for this. I am not sure.

Reply:
Can also use Optional on the CSRNonce, since bool + value is basically 'optional'. We use that for MDNS parameters.

Reply:
Made CSRNonce optional.